File: akka-http-core/src/main/java/akka/http/javadsl/model/AttributeKey.java
/*
 * Copyright (C) 2019-2025 Lightbend Inc. <https://akka.io>
 */

package akka.http.javadsl.model;

import akka.annotation.DoNotInherit;

@DoNotInherit
public abstract class AttributeKey<T> {
    public static <U> AttributeKey<U> create(String name, Class<U> clazz) {
        return new akka.http.scaladsl.model.AttributeKey<U>(name, clazz);
    }
}

File: akka-http-core/src/main/scala/akka/http/scaladsl/model/AttributeKey.scala
/*
 * Copyright (C) 2019-2022 Lightbend Inc. <https://www.lightbend.com>
 */

package akka.http.scaladsl.model

import akka.http.javadsl.{ model => jm }

import scala.reflect.ClassTag

case class AttributeKey[T](name: String, private val clazz: Class[_]) extends jm.AttributeKey[T]

object AttributeKey {
  def apply[T: ClassTag](name: String): AttributeKey[T] =
    new AttributeKey[T](name, implicitly[ClassTag[T]].runtimeClass)
}
